aioesphomeapi/client.py

```diff
@@ -468,25 +468,44 @@ def _on_home_assistant_service_response(
             (HomeassistantServiceResponse,),
         )
 
+    def _filter_bluetooth_message(
+        self,
+        address: int,
+        handle: int,
+        msg: message.Message,
+    ) -> bool:
+        """Handle a Bluetooth message."""
+        if TYPE_CHECKING:
+            assert isinstance(
+                msg,
+                (
+                    BluetoothGATTErrorResponse,
+                    BluetoothGATTNotifyResponse,
+                    BluetoothGATTReadResponse,
+                    BluetoothGATTWriteResponse,
+                ),
+            )
+        return bool(msg.address == address and msg.handle == handle)
+
     async def _send_bluetooth_message_await_response(
         self,
         address: int,
         handle: int,
         request: message.Message,
-        response_type: Type[message.Message],
+        response_type: Union[
+            Type[BluetoothGATTNotifyResponse],
+            Type[BluetoothGATTReadResponse],
+            Type[BluetoothGATTWriteResponse],
```

Comment:
I wish we could increase the minimum python version here as the new syntax is so much better